pypsi/core.py

```diff
@@ -266,6 +272,89 @@ def print_help(self, file=None):
         print(AnsiCodes.yellow, self.format_help(), AnsiCodes.reset, sep='',
               file=f)
 
+    def get_options(self):
+        '''
+        :return: All optional arguments
+        '''
+        return [key for key in self._op_callbacks]
+
+    def get_option_callback(self, arg):
+        return self._op_callbacks.get(arg, None)
+
+    def has_value(self, arg):
+        '''
+        Check if the optional argument has a value associated with it.
+        :param arg: Optional argument to check
+        :return: True if arg has a value, false otherwise
+        '''
+        val_actions = [argparse._AppendAction, argparse._StoreAction]
+        # Try catch in case arg doesn't exist
+        try:
+            # If a option's action is 'store' or 'append' is stores a value
+            return type(self._option_string_actions[arg]) in val_actions
+        except Exception as e:
+            print(e)
+            return False
+
+    def get_positional_callback(self, pos):
+        '''
```

Comment:
I was back and forth on this. Coming back to it, I agree, and have changed it to index 0.